### ISSUE TITLE
User file crashes with addition of SetpointManager when no actual node exists - Ticket 11155

### ISSUE THREAD

**Issue by rraustad** [NREL/EnergyPlus#5534]
## Issue overview

Please change this line to a description of the defect, with useful supporting information including how the issue was found, conditions that trigger this defect, workarounds for users of the current version, and a clear description of how to validate that a fix is complete for this defect.
### Details

Some additional details for this issue (if relevant):
- Platform (Operating system, version)
- Version of EnergyPlus (if using an intermediate build, include SHA)
### Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [ ] Defect file added (list location of defect file here)
- [ ] Ticket added to Pivotal for defect (development team task)
- [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by rraustad** [comment]
File crashes within this loop in function CheckNodeConnections in BranchNodeConnections.cc

 // Check 10 -- fluid streams cannot have multiple inlet/outlet nodes on same component
 // can have multiple inlets with one outlet or vice versa but cannot have multiple both inlet and outlet
 if ( NumOfNodeConnections > 0 ) {

Added last line here and program exited cleanly. There may be a better way to do this, outer IF check or something else.

 // NodeObjects now contains each consecutive object...
 for ( Object = 1; Object <= NumObjects - 1; ++Object ) {
 IsValid = true;
 FluidStreamInletCount = 0;
 FluidStreamOutletCount = 0;
 FluidStreamCounts = false;
 Loop1 = NodeObjects( Object );
 if ( NodeObjects.size() < 2 ) continue; // line 653 BranchNodeConnections.cc

**Comment by rraustad** [comment]
Revised method at line 653 to avoid crash:

```
if ( NumOfNodeConnections < 2 ) continue;
```

Correction provides appropriate severe warnings.

```
** Severe  ** Node Connection Error, Node="KNOTENZULUFT", Setpoint node did not find a matching node of appropriate type (other than Setpoint, OutsideAir).
**   ~~~   ** Reference Object=SETPOINTMANAGER:SCHEDULED, Name=ZULUFTSETPOINT
** Severe  ** Node Connection Error, Node="KNOTENZULUFT", Setpoint node did not find a matching node of type Inlet or Outlet.
**   ~~~   ** It appears this node is not part of the HVAC system.
**   ~~~   ** Reference Object=SETPOINTMANAGER:SCHEDULED, Name=ZULUFTSETPOINT
```

**Comment by rraustad** [comment]
Example file Minimal.idf runs in 8.4 (same appropriate severe warning as shown above) and crashes in 8.5 when these objects are added. This file, and the defect file, now run in this branch.

```
SetpointManager:Scheduled,
 ZuluftSetpoint,          !- Name
 MassFlowRate,            !- Control Variable
 StandardJahr,            !- Schedule Name
 KnotenZuluft;            !- Setpoint Node or NodeList Name

Schedule:Compact,
 StandardJahr,            !- Name
 Any Number,              !- Schedule Type Limits Name
 Through: 12/31,          !- Field 1
 For: AllDays,            !- Field 2
 Until: 24:00,131.8;      !- Field 3

ScheduleTypeLimits,
 Any Number;              !- Name
```

**Comment by rraustad** [comment]
See PR [#5624](https://github.com/NREL/EnergyPlus/pull/5624)

**Comment by Myoldmopar** [comment]
Closed with #5624
